internal/provider/waypoint/data_source_waypoint_application.go

```diff
@@ -211,5 +238,36 @@ func (d *DataSourceApplication) Read(ctx context.Context, req datasource.ReadReq
 		data.InputVars = types.SetNull(types.ObjectType{AttrTypes: InputVar{}.attrTypes()})
 	}
 
+	// Read the output values from the application and set them in the plan
+	diags := readAppOutputsData(ctx, application, &data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
 	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
 }
+
+func readAppOutputsData(ctx context.Context, app *waypoint_models.HashicorpCloudWaypointApplication, plan *ApplicationDataSourceModel) diag.Diagnostics {
```

Comment:
`readAppOutputsData` and `readAppOutputs` are identical save for one of their inputs (either a `*ApplicationDataSourceModel` or `*ApplicationResourceModel`, respectively. They also internalize the use of `diag.Diagnostics` and buries whatever happens when reading the outputs. What do you think of combining into a single method that just accepts a slice of `[]*HashicorpCloudWaypointTFOutputValue` and returning a slice of `[]*outputValue`, similar to what was done in https://github.com/hashicorp/terraform-provider-hcp/pull/864/files#diff-a94b22f215a353a3d43c5eb6fb4f8c9bc64d8c08eb9a4e7ef419eefefc7eae2fR853 ?

Reply:
That looks like a good improvement, will do!